test/test_start_date.py

```diff
@@ -88,15 +87,33 @@ def run_test(self, days, expected_streams):
 
                 # expected values
                 expected_primary_keys = self.expected_primary_keys()[stream]
-
                 # collect information for assertions from syncs 1 & 2 base on expected values
                 record_count_sync_1 = record_count_by_stream_1.get(stream, 0)
                 record_count_sync_2 = record_count_by_stream_2.get(stream, 0)
-
-                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                primary_keys_list_1 = [tuple(message.get('data').get(expected_pk)
+                                             for expected_pk in expected_primary_keys)
                                        for message in synced_records_1.get(stream, {}).get('messages', [])
                                        if message.get('action') == 'upsert']
-                primary_keys_list_2 = [tuple(message.get('data').get(expected_pk) for expected_pk in expected_primary_keys)
+                if self.expected_replication_method().get(stream) == 'INCREMENTAL' :
+                   expected_rk= self.expected_replication_keys().get(stream)
+                   expected_rk = expected_rk.pop()
+                   if stream == "tickets":
+                      primary_keys_list_2 = [tuple(message.get('data').get(expected_pk)
+                                          for expected_pk in expected_primary_keys)
+                                          for message in synced_records_2.get(stream, {}).get('messages', [])
+                                          if message.get('action') == 'upsert' and
+                                          self.parseunix_date(message.get('data').get(expected_rk)) <=
+                                             self.parseunix_date(last_record_date.get(stream).get(expected_rk))]
+                   else:
+                       primary_keys_list_2 = [tuple(message.get('data').get(expected_pk)
+                                            for expected_pk in expected_primary_keys)
+                                               for message in synced_records_2.get(stream, {}).get('messages', [])
+                                               if message.get('action') == 'upsert' and
+                                              self.parse_date(message.get('data').get(expected_rk)) <=
+                                              self.parse_date(last_record_date.get(stream).get(expected_rk))]
```

Comment:
There is probably a cool way to parameterize the method change here so that we can collapse these two loops into one and just call the right method for the stream although I'm not sure if this will work off the top of my head without testing maybe instead of the if at the top, after defining expected_rk try something like:
`parse_method = self.parse_unix_date if stream == tickets else self.parse_date`